File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionJsonParseTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import co.nlighten.jsontransform.adapters.gson.GsonHelpers;
import com.google.gson.Gson;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;

public class TransformerFunctionJsonParseTest extends BaseTest {

    Gson gson = GsonHelpers.GSON();

    @Test
    void fromString() {
        var strVal = "text";
        assertTransformation(gson.toJson(strVal), "$$jsonparse:$", strVal);
        assertTransformation(gson.toJson(strVal), "$$jsonparse():$", strVal);
    }
    @Test
    void fromBoolean() {
        var boolVal = true;
        assertTransformation(gson.toJson(boolVal), "$$jsonparse:$", boolVal);
    }
    @Test
    void fromNumber() {
        var numVal = 123;
        var str = gson.toJson(numVal);
        assertTransformation(str, "$$jsonparse:$", gson.fromJson(str, Object.class));
    }

    @Test
    void fromBigDecimal() {
        var numVal = new BigDecimal("1234567890.098765432123456789");
        var numStr = gson.toJson(numVal);
        assertTransformation(numStr, "$$jsonparse:$", gson.fromJson(numStr, Object.class));
        var bigVal = new BigDecimal("123456789123456789123456789123456789");
        var bigStr = gson.toJson(bigVal);
        assertTransformation(bigStr, "$$jsonparse:$", gson.fromJson(bigStr, Object.class));
    }

    @Test
    void fromObject() {
        var jsonString = "{\"a\":\"b\"}";
        var jsonAsObj = fromJson(jsonString);
        assertTransformation(jsonString, "$$jsonparse:$", jsonAsObj);
    }
    @Test
    void fromArray() {
        var jsonString = "[\"a\",\"b\"]";
        var jsonAsObj = fromJson(jsonString);
        assertTransformation(jsonString, "$$jsonparse:$", jsonAsObj);
    }

    @Test
    void objectFromObject() {
        var jsonString = "{\"a\":\"b\"}";
        var jsonAsObj = fromJson(jsonString);
        assertTransformation(jsonString, fromJson("""
{
"$$jsonparse": "$"
}"""), jsonAsObj);

    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionJsonParse.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";
import { BigDecimal, JSONBig } from "../../functions/common/FunctionHelpers";

describe("TransformerFunctionJsonParse", () => {
  test("fromString", async () => {
    const strVal = "text";
    await assertTransformation(JSON.stringify(strVal), "$$jsonparse:$", strVal);
    await assertTransformation(JSON.stringify(strVal), "$$jsonparse():$", strVal);
  });
  test("fromBoolean", async () => {
    const boolVal = true;
    await assertTransformation(JSON.stringify(boolVal), "$$jsonparse:$", boolVal);
  });
  test("fromNumber", async () => {
    const numVal = 123;
    const str = JSONBig.stringify(numVal);
    await assertTransformation(str, "$$jsonparse:$", JSONBig.parse(str));
  });

  test("fromBigDecimal", async () => {
    const numVal = BigDecimal("1234567890.098765432123456789");
    const numStr = JSONBig.stringify(numVal);
    await assertTransformation(numStr, "$$jsonparse:$", JSONBig.parse(numStr));
    const bigVal = BigDecimal("123456789123456789123456789123456789");
    const bigStr = JSONBig.stringify(bigVal);
    await assertTransformation(bigStr, "$$jsonparse:$", JSONBig.parse(bigStr));
  });

  test("fromObject", async () => {
    const jsonString = '{"a":"b"}';
    const jsonAsObj = JSON.parse(jsonString);
    await assertTransformation(jsonString, "$$jsonparse:$", jsonAsObj);
  });
  test("fromArray", async () => {
    const jsonString = '["a","b"]';
    const jsonAsObj = JSON.parse(jsonString);
    await assertTransformation(jsonString, "$$jsonparse:$", jsonAsObj);
  });

  test("objectFromObject", async () => {
    const jsonString = '{"a":"b"}';
    const jsonAsObj = JSON.parse(jsonString);
    await assertTransformation(
      jsonString,
      {
        $$jsonparse: "$",
      },
      jsonAsObj,
    );
  });
});
